gcs-action/src/main.ts

```diff
@@ -0,0 +1,61 @@
+import * as core from '@actions/core'
+import * as gcs from './gcs'
+async function run(): Promise<void> {
+  try {
+    const folderToUpload = core.getInput('folder-to-upload')
+    const fileToUpload = core.getInput('file-to-upload')
+    const bucketName = core.getInput('bucket-name')
+    const objectKeyPrefix = core.getInput('object-key-prefix')
+    const clearExistingFilesFirst: boolean =
+      core.getInput('clear-existing-files-first').toLowerCase() !== 'false'
+
+    if (folderToUpload && !fileToUpload) {
+      const uploadedFiles = await gcs.uploadDirectory(
+        bucketName,
+        folderToUpload,
+        objectKeyPrefix,
+        clearExistingFilesFirst
+      )
+
+      if (uploadedFiles) {
+        core.setOutput(
+          'uploadedSuccessFiles',
+          uploadedFiles
+            .filter(file => !(file.status instanceof Error))
+            .map(file => file.fileName)
+            .toString()
+        )
+        if (
+          uploadedFiles.filter(file => file.status instanceof Error).length > 0
```

Comment:
This _generally_ isn't the way errors and such are handled in Node land. However, if you have a compelling reason for doing it this way, I could be convinced.

Reply:
I guess my initial thoughts when approaching this action was to capture the files that errors instead of letting it bubble up. I think its cleaner to just let it bubble up like you suggested and set error in main